rllib/policy/tests/test_sample_batch.py

```diff
@@ -263,6 +263,22 @@ def test_slicing(self):
             },
         )
 
+    def test_split_by_episode(self):
+        s = SampleBatch(
+            {
+                "a": np.array([0, 1, 2, 3, 4, 5]),
+                "eps_id": np.array([0, 0, 0, 0, 1, 1]),
+                "dones": np.array([0, 0, 0, 1, 0, 1]),
+            }
+        )
+        eps_split = [b["a"] for b in s.split_by_episode()]
```

Comment:
Can you leave inline comments what is being tested in each section? 
Also can you modify your test logic to do the following?
1. check if your split_by_episode when episode exists returns the correct splits (hard-code the ground truth)
2. check if your split_by_done also returns the same thing (right now you check if it matches what ever split_by_eps_id returns. you don't check its absolute correctness.)
3. check if it properly raises error if neither done or eps_id exist
4. create a sample_batch where done is not properly set (all false or just false for the last portion), test if the behavior is expected.